src/components/member/MemberDetailsWizard/CustomForm/OtherDetails/OtherDetails.jsx

```diff
@@ -94,59 +103,102 @@ class OtherDetails extends React.Component {
     }
   }
 
+  /**
+   *  This method add and remove the disable summary schema and data on the
+   *  basis of isDisabled.
+   * @param {Object} formData
+   * @param {Object} formSchema
+   * @return {{formSchema: *, formData: *}}
+   */
+  generateSchemaForOtherDetails = (formData, formSchema) => {
+    const updatedFormData = formData;
+    let updatedFormSchema = cloneDeep(formSchema);
+    const {
+      differentlyAbledSummarySchema,
+      formData: { isDifferentlyAbled: prevIsDifferentlyAbled },
+    } = this.state;
+    const { isDifferentlyAbled } = formData;
+
+    // Checking form if schema have to update or not. If we update it unnecessarily
+    // then selected field looses focus.
+    const isFormSchemaUpdated = prevIsDifferentlyAbled !== isDifferentlyAbled;
+
+    if (isDifferentlyAbled) {
+      // if it is true
+      updatedFormSchema = {
+        ...updatedFormSchema,
+        properties: {
+          ...updatedFormSchema.properties,
+          differentlyAbledSummary: differentlyAbledSummarySchema,
+        },
+      };
+    } else {
+      // if it is false then removed the summary schema.
+      delete updatedFormSchema.properties.differentlyAbledSummary;
+      delete updatedFormData.differentlyAbledSummary;
+    }
+
+    return {
+      formData: updatedFormData,
+      formSchema: updatedFormSchema,
+      isFormSchemaUpdated,
+    };
+  };
+
   /**
    * In on change we are only check for isDifferentlyAbled flag if it is change
    * then update the schema.
    * @param {Object} formData
    */
   onChangeFormData = ({ formData }) => {
     const {
-      formData: { isDifferentlyAbled },
       formSchema,
-      differentlyAbledSummarySchema,
+      formDataConst,
+      formData: prevFormData,
     } = this.state;
 
-    let updatedFormSchema = formSchema;
-    const updatedFormData = formData;
 
-    const { isDifferentlyAbled: updatedisDifferentlyAbled } = formData;
-
-    // Checking if isDifferentlyAbled is changed
-    if (isDifferentlyAbled !== updatedisDifferentlyAbled) {
-      if (updatedisDifferentlyAbled) {
-        // if it is true
-        updatedFormSchema = {
-          ...updatedFormSchema,
-          properties: {
-            ...updatedFormSchema.properties,
-            differentlyAbledSummary: differentlyAbledSummarySchema,
-          },
-        };
-      } else {
-        // if it is false then removed the summary schema.
-        delete updatedFormSchema.properties.differentlyAbledSummary;
+    const {
+      formData: updatedFormData,
+      formSchema: updatedFormSchema,
+      isFormSchemaUpdated,
+    } = this.generateSchemaForOtherDetails(formData, formSchema);
 
-      }
+    // This code fragment is for checking if form data is dirty
+    const isFormDataEqual = isEqual(updatedFormData, prevFormData);
+    if (!isFormDataEqual) {
+      const isFormDataDirty = !isEqual(updatedFormData, formDataConst);
+      const isSubmitDisable = !isFormDataDirty;
```

Comment:
No need of storing it in isFormDataDirty variable
```suggestion
      const isSubmitDisabled = !isEqual(updatedFormData, formDataConst);
``` #

Reply:
done.